src/main/java/software/bernie/techarium/datagen/TechariumRecipeProvider.java

```diff
@@ -67,30 +72,57 @@ private void registerVanillaBotariumRecipes(Consumer<IFinishedRecipe> consumer)
     }
 
-    public static void buildBotariumRecipe(Item seed, IItemProvider drop, int amountWater, int time, Consumer<IFinishedRecipe> consumer) {
+    public static void buildBotariumRecipe(Item seed, ChancedItemStackList drop, int amountWater, int time, Consumer<IFinishedRecipe> consumer) {
         buildBotariumRecipe(seed, drop, Ingredient.of(TagRegistry.DIRT), amountWater, time, consumer);
     }
 
-    public static void buildBotariumRecipe(Item seed, IItemProvider drop, Ingredient soil, int amountWater, int time, Consumer<IFinishedRecipe> consumer) {
+    public static void buildBotariumRecipe(Item seed, ChancedItemStackList drop, Ingredient soil, int amountWater, int time, Consumer<IFinishedRecipe> consumer) {
         buildBotariumRecipe(seed, drop, soil, new FluidStack(Fluids.WATER, amountWater), time, consumer);
     }
 
-    private static void buildBotariumRecipe(Item seed, IItemProvider drop, Ingredient soil, FluidStack fluid, int time, Consumer<IFinishedRecipe> consumer) {
+    private static void buildBotariumRecipe(Item seed, ChancedItemStackList drop, Ingredient soil, FluidStack fluid, int time, Consumer<IFinishedRecipe> consumer) {
         BotariumRecipe.builder()
                 .cropType(Ingredient.of(seed))
                 .soilIn(soil)
                 .fluidIn(fluid)
                 .maxProgress(time)
                 .rfPerTick(10)
                 .progressPerTick(1)
-                .output(new ItemStack(drop, 1))
+                .output(drop)
                 .construct()
                 .build(consumer,
                         new ResourceLocation(Techarium.ModID,
                                 "botarium/minecraft/" + seed.getRegistryName().getPath()));
     }
+
+    public void buildArboretumRecipe(Item sapling, ChancedItemStackList drop, int amountWater, int time, Consumer<IFinishedRecipe> consumer) {
+        buildArboretumRecipe(sapling, drop, Ingredient.of(TagRegistry.DIRT), amountWater, time, consumer);
+    }
+
+    public void buildArboretumRecipe(Item sapling, ChancedItemStackList drop, Ingredient soil, int amountWater, int time, Consumer<IFinishedRecipe> consumer) {
+        buildArboretumRecipe(sapling, drop, soil, new FluidStack(Fluids.WATER, amountWater), time, consumer);
+    }
+
+    private void buildArboretumRecipe(Item sapling, ChancedItemStackList drop, Ingredient soil, FluidStack fluid, int time, Consumer<IFinishedRecipe> consumer) {
+        ArboretumRecipe.builder()
+                .cropType(Ingredient.of(sapling))
+                .soilIn(soil)
+                .fluidIn(fluid)
+                .maxProgress(time)
+                .rfPerTick(10)
+                .progressPerTick(1)
+                .output(drop)
+                .construct()
+                .build(consumer,
+                        new ResourceLocation(Techarium.ModID,
+                                "arboretum/minecraft/" + sapling.getRegistryName().getPath()));
```

Comment:
Could we make this go in a folder based on the sapling's resource location's namespace? So minecraft saplings go in a /minecraft/ folder, thermal go in /thermal/, etc. The botarium recipes would be a good example for what I mean